src/media/modules/class-wpml-elementor-media-node-gallery.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+class WPML_Elementor_Media_Node_Gallery extends \WPML_Elementor_Media_Node {
```

Comment:
@kkarpieszuk, I guess we already abstracted this data shape. Did you try this?

```php
namespace WPML\PB\Elementor\Media\Modules;
    
class Gallery extends \WPML_Elementor_Media_Node_With_Images_Property {
    protected function get_property_name() {
        return 'gallery';
    }
}
```

(please use a namespace for the new classes)

Reply:
@kkarpieszuk, I can't see what I suggested in your latests commits.

Also, please rename the file `Gallery.php`.

Reply:
@strategio file renamed. 

But I didn't get what you are asking for here. To introduce `get_property_name()` method? I don't need it, all is needed here is `translate()` method which looks for `gallery` array element and for each found, translate each internal image array

Reply:
wait... I think I understand now: if I declare class as extending `WPML_Elementor_Media_Node_With_Images_Property` (I didn't notice it) all I might need is to define property name, got it. Let me try...

Reply:
@strategio nope, it doesn't work, seems like this kind of element does not have `WPML_Page_Builders_Media_Translate` structure same as the rest which are translated with `WPML_Elementor_Media_Node_With_Images_Property`